src/cocaine-app/sched/ttl_cleanup_starter.py

```diff
@@ -0,0 +1,119 @@
+import logging
+
+from infrastructure import infrastructure
+import jobs
+from mastermind_core.config import config
+import storage
+import time
+import datetime
+from yt_worker import YqlWrapper
+
+
+logger = logging.getLogger('mm.sched.ttl_cleanup')
+
+class TtlCleanupStarter(object):
+    def __init__(self, scheduler):
+        scheduler.register_periodic_func(self._do_ttl_cleanup,
+                                         period_default=60*15,  # 15 minutes
+                                         starter_name="ttl_cleanup")
+        self.params = config.get('scheduler', {})
+        self.scheduler = scheduler
+
+    def _get_yt_stat(self):
+        """
+        Extract statistics from YT logs
+        :return: list of couples ids with expired records volume above specified
+        """
+        try:
+            # Configure parameters for work with YT
+            yt_cluster = self.params.get('ttl_cleanup', {}).get('yt_cluster', "")
+            yt_token = self.params.get('ttl_cleanup', {}).get('yt_token', "")
+            yt_attempts = self.params.get('ttl_cleanup', {}).get('yt_attempts', 3)
+            yt_delay = self.params.get('ttl_cleanup', {}).get('yt_delay', 10)
+
+            yt_wrapper = YqlWrapper(cluster=yt_cluster, token=yt_token, attempts=yt_attempts, delay=yt_delay)
+
+            aggregation_table = self.params.get('ttl_cleanup', {}).get('aggregation_table', "")
+            base_table = self.params.get('ttl_cleanup', {}).get('tskv_log_table', "")
+            expired_threshold = self.params.get('ttl_cleanup', {}).get('ttl_threshold', 10 * float(1024 ** 3))  # 10GB
+
+            yt_wrapper.prepare_aggregate_for_yesterday(base_table, aggregation_table)
+
+            couple_list = yt_wrapper.request_expired_stat(aggregation_table, expired_threshold)
+            logger.info("YT request has completed")
+            return couple_list
```

Comment:
fixed in https://github.com/yandex/mastermind/pull/434/commits/968d0fbf32e9a25f5713cb1d2e5c4d336cf6a401